ctapipe/reco/tests/test_HillasReconstructor.py

```diff
@@ -72,154 +83,260 @@ def test_h_max_results():
-            except HillasParameterizationError:
+                moments_camera_frame = hillas_parameters(
+                    geom_camera_frame[mask], dl1.image[mask]
+                )
+                moments_telescope_frame = hillas_parameters(
+                    geom_telescope_frame[mask], dl1.image[mask]
+                )
+
+                if (moments_camera_frame.width.value > 0) and (moments_telescope_frame.width.value > 0):
+                    event_camera_frame.dl1.tel[
+                        tel_id
+                    ].parameters.hillas = moments_camera_frame
+                    dl1.parameters.hillas = moments_telescope_frame
+
+                    hillas_dict_camera_frame[tel_id] = moments_camera_frame
+                    hillas_dict_telescope_frame[tel_id] = moments_telescope_frame
+                else:
+                    continue
+
+            except HillasParameterizationError as e:
+                print(e)
                 continue
 
-        # construct a dict only containing the last telescope events
-        # (#telescopes < 2)
-        hillas_dict_only_one_tel = dict()
-        hillas_dict_only_one_tel[tel_id] = hillas_dict[tel_id]
-        with pytest.raises(TooFewTelescopesException):
-            fit.predict(hillas_dict_only_one_tel, subarray, tel_azimuth, tel_altitude)
-
-        # construct a hillas dict with the width of the last event set to 0
-        # (any width == 0)
-        hillas_dict_zero_width = hillas_dict.copy()
-        hillas_dict_zero_width[tel_id]["width"] = 0 * u.m
-        with pytest.raises(InvalidWidthException):
-            fit.predict(hillas_dict_zero_width, subarray, tel_azimuth, tel_altitude)
-
-        # construct a hillas dict with the width of the last event set to np.nan
-        # (any width == nan)
-        hillas_dict_nan_width = hillas_dict.copy()
-        hillas_dict_zero_width[tel_id]["width"] = np.nan * u.m
-        with pytest.raises(InvalidWidthException):
-            fit.predict(hillas_dict_nan_width, subarray, tel_azimuth, tel_altitude)
+        if (len(hillas_dict_camera_frame) > 2) and (len(hillas_dict_telescope_frame) > 2):
+            reconstructor(event_camera_frame)
+            reconstructor(event)
+            reconstructed_events += 1
+        else:  # this event was not good enough to be tested on
+            continue
+
+        # Compare old approach with new approach
+        result_camera_frame = event_camera_frame.dl2.stereo.geometry["HillasReconstructor"]
+        result_telescope_frame = event.dl2.stereo.geometry["HillasReconstructor"]
+
+        assert result_camera_frame.is_valid
+        assert result_telescope_frame.is_valid
+
+        for field in event.dl2.stereo.geometry["HillasReconstructor"].as_dict():
+            C = np.asarray(result_camera_frame.as_dict()[field])
+            T = np.asarray(result_telescope_frame.as_dict()[field])
+            assert (np.isclose(C, T, rtol=1e-03, atol=1e-03, equal_nan=True)).all()
```

Comment:
I think we use `np.testing.assert_allclose` for assertions on arrays:

Reply:
from https://numpy.org/doc/stable/reference/generated/numpy.isclose.html
I read "Returns a boolean array where **two arrays** are **element-wise** equal within a tolerance."